src/flb_oauth2.c

```diff
@@ -115,6 +115,7 @@ int flb_oauth2_parse_json_response(const char *json_data, size_t json_size,
         }
         else if (key_cmp(key, key_len, "expires_in") == 0) {
             ctx->expires_in = atol(val);
+            ctx->expires = time(NULL) + ctx->expires_in;
```

Comment:
The description of the `expires` field in `struct flb_oauth2` [says "token times set by the caller"](https://github.com/fluent/fluent-bit/blob/37aa680d32384c1179f02ee08a5bef4cd278513e/include/fluent-bit/flb_oauth2.h#L40). This function simply returns the `access_token`, `token_type`, and `expires_in` values parsed from the JSON response. Does it make sense to set `expires` in line 394 (after a successful call to `flb_oauth2_parse_json_response`) instead?

Mildly related aside: should we be setting `issued` as well? It doesn't seem to be used anywhere…

Reply:
Thanks for reviewing it. I have updated both of them.